project/src/main/scala/ScalastyleInspections.scala

```diff
@@ -16,6 +16,9 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  */
+package com.mwz.sonar.scala
+package scalastyle
+
 import org.scalastyle._
 
 case class ScalastyleInspection(
```

Comment:
I'm sorry I did not notice it before...
But, we should make those two case classes `final`, and also we should make them only visible in the scalastyle package.

However, as you said before this needs refactor and you want to see how this behave first. So, we could omit this for now.

Reply:
> "funny how rarely you see people make their case classes final."

I think Scala 3, will make them final by default.

> "The problem is that probably the recommended quality profile code is going to live on the same level as scalastyle and scapegoat, so if we want to have access to those inspections there, we won't be able to make them private to their own packages..."

Ah, I see, well nothing to do then.